service/matching/task_queue_partition_manager.go

```diff
@@ -517,6 +517,53 @@ func (pm *taskQueuePartitionManagerImpl) callerInfoContext(ctx context.Context)
 	return headers.SetCallerInfo(ctx, headers.NewBackgroundCallerInfo(pm.ns.Name().String()))
 }
 
+// ForceLoadAllNonRootPartitions spins off go routines which make RPC calls to all the
+func (pm *taskQueuePartitionManagerImpl) ForceLoadAllNonRootPartitions(ctx context.Context) {
+	if !pm.partition.IsRoot() {
+		pm.logger.Info("ForceLoadAllNonRootPartitions called on non-root partition. Prevented circular keep alive (loading) of partitions.")
```

Comment:
FYI, this isn't the only way that this could cause circular keep-alive. The more interesting situation is:
1. root loads and force-loads child
2. child starts goroutine to call GetTaskQueueUserData on root periodically
3. root idle times out and is unloaded
4. child then happens to call GetTaskQueueUserData
5. root gets loaded again and force-loads child, causing child to stay loaded longer
That can't happen because in 2, GetTaskQueueUserData resets the idle timer, so 3 can't happen. And also in 5, force-loading child does _not_ reset the idle timer. But that's the one to be aware of.